src/components/conversationalsearchresult/conversationalsearchresult.js

```diff
@@ -0,0 +1,164 @@
+    this.reduxStore = reduxStore;
+    this.answerMaxHeight = conf.answerMaxHeight || 150;
+    this.isSearchResultLoading = false;
+    this.observeResultLoadingState();
+
+    registerHelper('markdown', function (text) {
+      if (typeof text !== 'string') {
+        return '';
+      }
+
+      return new handlebars.SafeString(marked.parse(text, { breaks: true, gfm: true }));
+    });
+
+    if (validateContainer(conf.containerId)) {
+      observeStoreByKey(this.reduxStore, 'search', () => this.render());
+    }
+  }
+
+  observeResultLoadingState() {
+    this.reduxStore.subscribe(() => {
+      const isConversationalSearchResultLoading =
+        this.reduxStore.getState().search.loadingConversationalSearchResult;
+
+      if (this.isSearchResultLoading !== isConversationalSearchResultLoading) {
+        this.isSearchResultLoading = isConversationalSearchResultLoading;
+        this.render();
+      }
+    });
+  }
+
+  setupShowMoreButton() {
+    const answerContainer = document.querySelector('.answer-container');
+    const fadeOutOverlay = answerContainer.querySelector('.fade-out-overlay');
+    const showMoreBtn = document.querySelector('.show-more-btn');
+    const buttonText = showMoreBtn.querySelector('.button-text');
+    const chevron = showMoreBtn.querySelector('.chevron');
+
+    if (answerContainer && showMoreBtn) {
+      answerContainer.style.maxHeight = `${this.answerMaxHeight}px`;
+
+      if (answerContainer.scrollHeight > this.answerMaxHeight) {
+        showMoreBtn.style.display = 'flex';
+        fadeOutOverlay.style.display = 'block';
+      } else {
+        showMoreBtn.style.display = 'none';
+        fadeOutOverlay.style.display = 'none';
+      }
+
+      showMoreBtn.addEventListener('click', () => {
+        if (answerContainer.classList.contains('collapsed')) {
+          answerContainer.style.maxHeight = `${answerContainer.scrollHeight}px`;
+          answerContainer.classList.remove('collapsed');
+          buttonText.textContent = 'Show less';
+          fadeOutOverlay.style.display = 'none';
+          chevron.style.transform = 'rotate(180deg)';
+        } else {
+          answerContainer.style.maxHeight = `${this.answerMaxHeight}px`;
+          answerContainer.classList.add('collapsed');
+          buttonText.textContent = 'Show more';
+          fadeOutOverlay.style.display = 'block';
+          chevron.style.transform = '';
+        }
+      });
+    }
+  }
+
+  setupActionButtons() {
+    const conversationalSearchResultContainer = document.querySelector(
+      '.addsearch-conversational-search-result'
+    );
+    const copyButton = conversationalSearchResultContainer.querySelector('.copy-btn');
+    const copyConfirm = conversationalSearchResultContainer.querySelector('.copy-confirm-message');
+    const thumbsUpButton = conversationalSearchResultContainer.querySelector('.thumbs-up-btn');
+    const thumbsDownButton = conversationalSearchResultContainer.querySelector('.thumbs-down-btn');
```

Comment:
Query selectors and event listener should be checked for element validity, these CSS selector stays true only when using the default template.
When use custom template with different CSS class names, the app is broken

Reply:
true